src/main/java/ch/njol/skript/sections/SecWhile.java

```diff
@@ -117,7 +222,16 @@ public TriggerItem getActualNext() {
 
 	@Override
 	public String toString(@Nullable Event event, boolean debug) {
-		return (doWhile ? "do " : "") + "while " + condition.toString(event, debug);
+		SyntaxStringBuilder builder = new SyntaxStringBuilder(event, debug);
+		if (doWhile && state != WhileState.DO)
```

Comment:
```suggestion
		if (doWhile)
```
Technically not necessary since `state == WhileDate.DO` implies `!doWhile`

Reply:
Not necessarily, just as I transfer over the conditions from the  multilined all/any, I also transfer over the doWhile

Reply:
Ahh that's fair. Disregard this.